### ISSUE TITLE
Generalized for OpenID Connect

### ISSUE THREAD

**Issue by sdoxsee** [zquestz/omniauth-google-oauth2#200]
As Google follows the OpenID Connect spec, could the Provider URL and endpoints, etc. configurable so that this Gem can be used for other OpenID Connect Providers? https://github.com/jjbohn/omniauth-openid-connect has some of this functionality but is buggy and has been quite stagnant as far as releases go.

**Comment by sdoxsee** [comment]
I generalized the omniauth-google-oauth2 gem for openid connect (removing google provider specifics). I removed all the specs and it's desperately lacking in documentation but it's working for me for non-google providers. Here it is: https://github.com/sdoxsee/omniauth-oidc. Would be great if omniauth-google-oauth2 still had google defaults but could be configured to use a different openid connect provider.
